src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/products/ProductReferenceResolverIT.java

```diff
@@ -167,13 +167,18 @@ void sync_withNewProductWithNoProductTypeKey_ShouldFailCreatingTheProduct() {
 
         // assertion
         assertThat(syncStatistics).hasValues(1, 0, 0, 1);
-        assertThat(errorCallBackMessages).containsExactly(format("Failed to resolve references on ProductDraft with"
-                + " key:'%s'. Reason: Failed to resolve 'product-type' resource identifier on ProductDraft with "
-                + "key:'%s'. Reason: %s",
-            productDraft.getKey(), productDraft.getKey(),
-            BLANK_ID_VALUE_ON_RESOURCE_IDENTIFIER));
         assertThat(errorCallBackExceptions).hasSize(1);
-        assertThat(errorCallBackExceptions.get(0)).isExactlyInstanceOf(ReferenceResolutionException.class);
+        if (errorCallBackExceptions.get(0) instanceof ReferenceResolutionException) {
```

Comment:
a test assertion should not expect it with if, you need to expect it directly, please add additional tests for different cases.

Reply:
The if-else block has been removed in latest change.